### ISSUE TITLE
Add Gitlab Issues integration

### ISSUE THREAD

**Issue by xnutsive** [gravitational/teleport-plugins#70]
### Goal

Add a Gitlab integration that would work similarly to Jira Cloud flow. The user has a board, new requests pop up as issues on the board, and users can easily approve/deny them by dragging them to another column.

### Research

* Gitlab doesn’t seem to have “workflows” like Jira had. They have Boards, but those boards have columns (lists) that are based on 1) status: open or closed, 2) assignee, 3) label. You can’t set board logic to change Issue status from open to closed when it’s dragged into “Done” column. 
If we want to have a visual dashboard for requests, we can create a board automatically, and then place an issue on the board. 
* We can use Labels + Webhooks to approve/deny request. Label “Teleport: Approved” vs “Teleport: Denied” for example. 
* Gitlab doesn’t seem to have custom metadata fields in Issues, so we’ll have to store teleport_request_id in the first comment on the issue, or in the issue body, and regexp it ##out.

### Workflow

1. When a new request is created, our bot will create a new issue in Gitlab and add a label Teleport: Request to it. 
2. The user sees the Board and a new isssue in the Requests column. Opens the issue, and sets the label to Teleport: Approved, for example. 
3. Webhook notifies our bot. The bot reads the label, approves the request, and closes the issue.
As an enhancement, we can automate the whole setup process with the API, specifically, we’ll need to create:
* Three labels: Teleport: Approved, Teleport: Declined and Teleport: Request
* A new Board with 3 columns for said labels. 

**UPD**

- I forgot to check if Gitlab provides WebHooks for Issue events. [It does, here's the link to their docs](https://docs.gitlab.com/ee/user/project/integrations/webhooks.html). 
- We can [manage webhooks via their API](https://docs.gitlab.com/ee/api/projects.html#add-project-hook).
- API support [personal API tokens](https://docs.gitlab.com/ee/api/README.html#personal-access-tokens) for authorization — that's pretty easy.

**Comment by xnutsive** [comment]
@benarent, @marshall-lee, I've just finally got to testing the plugin! Everything worked from the first try, and was very easy to setup. Vlad, thank you for working on this, and for the instructions. The plugin even sets up the webhooks and labels automatically 👏 

I tried merging `master` into this branch to have all the updates, but couldn't do that easily — @marshall-lee updated `utils.HTTP` and it's config. I'll leave the merge to Vlad then. 

@benarent @stevenGravy, I've also only written down a minimal set of instructions, as copying and pasting stuff from other readme files felt just wrong, and Steven is rewriting my crappy readmes anyway ;)
